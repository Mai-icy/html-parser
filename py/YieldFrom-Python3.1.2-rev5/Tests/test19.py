#
#   Test attempted yield-from loop
#

def g1():
	print("g1: starting")
	yield "y1"
	print("g1: about to yield from g2")
	yield from g2()
	print("g1 should not be here")

def g2():
	print("g2: starting")
	yield "y2"
	print("g2: about to yield from g1")
	yield from gi
	print("g2 should not be here")

gi = g1()
for y in gi:
	print("Yielded:", y)
