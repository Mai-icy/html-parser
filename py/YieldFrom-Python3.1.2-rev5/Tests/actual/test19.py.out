g1: starting
Yielded: y1
g1: about to yield from g2
g2: starting
Yielded: y2
g2: about to yield from g1
Traceback (most recent call last):
  File "test19.py", line 20, in <module>
    for y in gi:
  File "test19.py", line 16, in g2
    yield from gi
  File "test19.py", line 9, in g1
    yield from g2()
ValueError: generator already executing
